cmd/templates/http.go

```diff
@@ -51,15 +53,15 @@ func handleHTTPServer(ctx context.Context, u *url.URL, templatesEndpoints *templ
 	// Wrap the multiplexer with additional middlewares. Middlewares mounted
 	// here apply to all the service endpoints.
 	var handler http.Handler = mux
-	{
-		handler = httpMiddleware.Log(middleware.NewLogger(logger))(handler)
-		handler = httpMiddleware.RequestID()(handler)
-		handler = dataDog.WrapHandler(handler, "templates-api", "")
-	}
+	handler = httpMiddleware.Log(middleware.NewLogger(logger))(handler)
+	handler = httpMiddleware.RequestID()(handler)
+	handler = dataDog.WrapHandler(handler, "templates-api", "")
 
 	// Start HTTP server using default configuration, change the code to
 	// configure the server as required by your service.
-	srv := &http.Server{Addr: u.Host, Handler: handler}
+	requestCtx := context.WithValue(context.Background(), dependencies.CtxKey, d)
+	ctxProvider := func(_ net.Listener) context.Context { return requestCtx }
+	srv := &http.Server{Addr: u.Host, Handler: handler, BaseContext: ctxProvider}
```

Comment:
Kazdy request dostane `ctx`, ktory doputuje az ku nasmu `endpoint`u, ... cheme aby v nom boli **dependencies**.

Nie je to napojene na `ctx`, ktory pouziva server.
Pretoze, ked sa zavola `cancelFn`, tak hlavny `ctx` konci, a sever sa `30s` vypina.
No a mi chceme aby requesty tiez dobehli, a nie aby sa hned ukoncili.